src/dune_rules/install_rules.ml

```diff
@@ -205,7 +205,12 @@ end = struct
         let visibility = Module.visibility m in
         let dir' = Obj_dir.cm_dir external_obj_dir cm_kind visibility in
         if Path.equal (Path.build dir) dir' then None
-        else Path.basename dir' |> Option.some
+        else
+          String.drop_prefix (Path.to_string dir')
+            ~prefix:(Path.Build.to_string dir)
+          |> Option.value_exn
+          |> String.drop_prefix_if_exists ~prefix:"/"
```

Comment:
The fix is fine, but this little snippet of path handling code has been duplicated a couple of times already. Would you mind extracting it into a proper path handling function? It can be done in this PR with an additional commit, or in a separate PR.

Reply:
fixed this in the latest push